### DISCUSSION TITLE
create/show main window if user attempts to launch second instance

### DISCUSSION THREAD

**Discussion by KroniK907** [tauri-apps/tauri#8871] [Q&A]
Hello,

Lets say that the user has opened a Tauri app in a linux desktop and has closed the main app window, but the app is still running in the background, and a new window could be opened with the system tray icon. Later, the user forgets that the app is running in the background and attempts to launch the app again. How would I prevent the app from trying to launch a second instance and instead create a new window for the already running instance?

**Comment by FabianLars** [comment]
Sounds like you're looking for https://github.com/tauri-apps/plugins-workspace/tree/v1/plugins/single-instance. It will send an event to the primary instance that you can handle however you want (like showing and focusing the main window)
